app/scripts/controllers/onboarding.ts

```diff
@@ -0,0 +1,101 @@
+import { ObservableStore } from '@metamask/obs-store';
+import log from 'loglevel';
+
+type flowTypes = 'create' | 'import';
+/**
+ * The state of the OnboardingController
+ */
+export type OnboardingControllerState = {
+  seedPhraseBackedUp: boolean;
+  firstTimeFlowType: flowTypes | null;
+  completedOnboarding: boolean;
+  onboardingTabs: Record<string, string>;
+};
+
+const defaultTransientState: Record<string, Record<string, string>> = {
+  onboardingTabs: {},
+};
+
+const defaultState: OnboardingControllerState = {
+  seedPhraseBackedUp: false,
+  firstTimeFlowType: null,
+  completedOnboarding: false,
+  onboardingTabs: {},
+};
+
+/**
+ * Controller responsible for maintaining
+ * state related to onboarding
+ */
+export default class OnboardingController {
+  /**
+   * Observable store containing controller data.
+   */
+  store: ObservableStore<OnboardingControllerState>;
+
+  /**
+   * Constructs a Onboarding  controller.
+   *
+   * @param options - the controller options
+   * @param options.state - Initial controller state.
+   */
+  constructor({ state }: { state: OnboardingControllerState }) {
+    this.store = new ObservableStore({
+      ...defaultState,
+      ...state,
+      ...defaultTransientState,
+    });
+  }
+
+  /**
+   * Setter for the `seedPhraseBackedUp` property
+   *
+   * @param newSeedPhraseBackUpState - Indicates if the seedphrase is backup by the user or not
+   */
+  setSeedPhraseBackedUp(newSeedPhraseBackUpState: boolean) {
+    this.store.updateState({
+      seedPhraseBackedUp: newSeedPhraseBackUpState,
+    });
+  }
+
+  /**
+   * Sets the completedOnboarding state to true, indicating that the user has completed the
+   * onboarding process.
+   */
+  async completeOnboarding(): Promise<boolean> {
```

Comment:
Can this one be simplified as `<boolean>` here?

Reply:
Since it is an `async` function, it is expecting `Promise` in the return type.